src/pts/pyspark/panel_app.py

```diff
@@ -0,0 +1,305 @@
+    PHENOTYPE_BEFORE_SPLIT_RE = {
+        # Fixes for specific records.
+        r'\(HP:0006574;\);': r'(HP:0006574);',
+        r'Abruzzo-Erickson;syndrome': r'Abruzzo-Erickson syndrome',
+        r'Deafness, autosomal recessive; 12': r'Deafness, autosomal recessive, 12',
+        r'Waardenburg syndrome, type; 3': r'Waardenburg syndrome, type 3',
+        r'Ectrodactyly, ectodermal dysplasia, and cleft lip/palate syndrome; 3': (
+            r'Ectrodactyly, ectodermal dysplasia, and cleft lip/palate syndrome, 3'
+        ),
+        # Remove curly braces. They are *sometimes* (not consistently) used to separate disease name and OMIM code, for
+        # example: "{Bladder cancer, somatic}, 109800", and interfere with regular expressions for extraction.
+        r'[{}]': r'',
+        # Fix cases like "Aarskog-Scott syndrome, 305400Mental retardation, X-linked syndromic 16, 305400", where
+        # several phenotypes are glued to each other due to formatting errors.
+        r'(\d{6})([A-Za-z])': r'$1;$2',
+        # Replace all tab/space sequences with a single space.
+        r'[\t ]+': r' ',
+        # Remove leading and/or trailing spaces around semicolons.
+        r' ?; ?': r';',
+    }
+
+    # Cleanup regular expressions which are applied to the phenotypes *after* splitting.
+    PHENOTYPE_AFTER_SPLIT_RE = (
+        r' \(no OMIM number\)',
+        r' \(NO phenotype number in OMIM\)',
+        r'(no|No|NO) OMIM( phenotype|number|entry|NUMBER|NUMBER OR DISEASE)?',
+        r'[( ]*(from )?PMID:? *\d+[ ).]*',
+    )
+
+    # Regular expressions for extracting ontology information.
+    LEADING = r'[ ,-]*'
+    SEPARATOR = r'[:_ #]*'
+    TRAILING = r'[:.]*'
+    OMIM_RE = LEADING + r'(OMIM|MIM)?' + SEPARATOR + r'(\d{6})' + TRAILING
+    OTHER_RE = LEADING + r'(OrphaNet: ORPHA|Orphanet|ORPHA|HP|MONDO)' + SEPARATOR + r'(\d+)' + TRAILING
+
+    # Regular expression for filtering out bad PMID records.
+    PMID_FILTER_OUT_RE = r'224,614,752,030,146,000,000,000'
+
+    # Regular expressions for extracting publication information from the API raw strings.
+    PMID_RE = [
+        (
+            # Pattern 1, e.g. '15643612' or '28055140, 27333055, 23063529'.
+            r'^'  # Start of the string.
+            r'[\d, ]+'  # A sequence of digits, commas and spaces.
+            r'(?: |$)'  # Ending either with a space, or with the end of the string.
+        ),
+        (
+            # Pattern 2, e.g. '... observed in the patient. PMID: 1908107 - publication describing function of ...'
+            r'(?:PubMed|PMID)'  # PubMed or a PMID prefix.
+            r'[: ]*'  # An optional separator (spaces/colons).
+            r'[\d, ]+'  # A sequence of digits, commas and spaces.
+        ),
+    ]
+
+    def __init__(self, spark: Session, panelapp_df: DataFrame):
+        self.spark = spark
+        self.panelapp_df = panelapp_df
+
+    def generate_panelapp_evidence(self, panels) -> DataFrame:
```

Comment:
If `panels` is a mandatory requirement, I'd add it as an attribute to `PanelAppEvidenceGenerator` (so it's stored in the class initialisation).

If it's not mandatory -which I don't think it is-, I'd make this argument optional and add the information with an if block.